### DISCUSSION TITLE
Why APGL license?

### DISCUSSION THREAD

**Discussion by baobabresearch** [mastodon/mastodon#24581] [General]
Why Mastodon is under AGPL license? If I want to use it for commercial projects, I can't do that. 

I wanted to make a corporate social network for my employees, and with a closed code (for security purposes), but in the end I would have to "reinvent the wheel".

I don't see this as a complete open-source project, so I don't understand what's benefits of "open-source" there. I would donate to this project but I will not, sorry. 

Btw, Apache/MIT licenses is more freely.

**Comment by nightpool** [comment]
you can always use AGPL licensed code for commercial projects as long as you make the source code open source. The question answers itself

**Reply by nightpool** [reply]
> I wanted to make a corporate social network for my employees, and with a closed code

> more freely

You can pick one of two things—closed source code, or freedom. The project has decided on one it prefers.

**Comment by fieldofdisarray** [comment]
> Why Mastodon is under AGPL license? If I want to use it for commercial projects, I can't do that.
> 
> I wanted to make a corporate social network for my employees, and with a closed code (for security purposes), but in the end I would have to "reinvent the wheel".
> 
> I don't see this as a complete open-source project, so I don't understand what's benefits of "open-source" there. I would donate to this project but I will not, sorry.
> 
> Btw, Apache/MIT licenses is more freely.

lol.
